docs/operations/debug.md

```diff
@@ -91,4 +116,107 @@ $> kubectl logs -n <KubeVirt Install Namespace> virt-handler-2m86x | head -n8
 ```
 
 Obviously, for both examples above, `<KubeVirt Install Namespace>` needs to be replaced with the actual namespace
-KubeVirt is installed in.
\ No newline at end of file
+KubeVirt is installed in.
+
+
+## KubeVirt PProf Profiler
+Using the `cluster-profiler` client tool, a developer can get the PProf profiling data for every component in the Kubevirt Control plane. Here is a user guide:
+
+### Compile `cluster-profiler`
+Build from source code
+```
+git clone https://github.com/kubevirt/kubevirt.git
+cd kubevirt/tools/cluster-profiler
+go build
+```
+
+### Enable the feature gate
+Add `ClusterProfiler` in KubeVirt config
+```
+cat << END > enable-feature-gate.yaml
+---
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+        - ClusterProfiler
+END
+
+kubectl apply -f enable-feature-gate.yaml
+```
+
+### Do the profiling
+Start CPU profiling
+```
+$ cluster-profiler --cmd start
+2023/05/17 09:31:09 SUCCESS: started cpu profiling KubeVirt control plane
+```
+Stop CPU profiling
+```
+$ cluster-profiler --cmd stop
+2023/05/17 09:31:14 SUCCESS: stopped cpu profiling KubeVirt control plane
+```
+Dump the pprof result
+```
+$ cluster-profiler --cmd dump
+2023/05/17 09:31:18 Moving already existing "cluster-profiler-results" => "cluster-profiler-results-old-67fq"
+SUCCESS: Dumped PProf 6 results for KubeVirt control plane to [cluster-profiler-results]
+```
+The PProf result can be found in the folder `cluster-profiler-results`
+```
+tree cluster-profiler-results
```

Comment:
```suggestion
$ tree cluster-profiler-results

```

For consistency, whenever you are providing command examples with their output, include the prompt character (and separate command from output with a newline as mentioned above).

For blocks instructing the user to run a command, leave the prompts and the output off.